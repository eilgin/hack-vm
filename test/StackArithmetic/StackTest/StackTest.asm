@JUMP_CMP_TRUE
0;JMP
(IF_CMP_TRUE)
@SP
A=M
M=-1
@R15
A=M
0;JMP
(JUMP_CMP_TRUE)
@17
D=A
@SP
AM=M+1
A=A-1
M=D
@17
D=A
@SP
AM=M+1
A=A-1
M=D
@EQ_0
D=A
@R15
M=D
@SP
AM=M-1
D=M
@SP
AM=M-1
D=M-D
@IF_CMP_TRUE
D;JEQ
@SP
M=0
(EQ_0)
@SP
M=M+1
@892
D=A
@SP
AM=M+1
A=A-1
M=D
@891
D=A
@SP
AM=M+1
A=A-1
M=D
@LT_0
D=A
@R15
M=D
@SP
AM=M-1
D=M
@SP
AM=M-1
D=M-D
@IF_CMP_TRUE
D;JLT
@SP
M=0
(LT_0)
@SP
M=M+1
@32767
D=A
@SP
AM=M+1
A=A-1
M=D
@32766
D=A
@SP
AM=M+1
A=A-1
M=D
@GT_0
D=A
@R15
M=D
@SP
AM=M-1
D=M
@SP
AM=M-1
D=M-D
@IF_CMP_TRUE
D;JGT
@SP
M=0
(GT_0)
@SP
M=M+1
@56
D=A
@SP
AM=M+1
A=A-1
M=D
@31
D=A
@SP
AM=M+1
A=A-1
M=D
@53
D=A
@SP
AM=M+1
A=A-1
M=D
@SP
AM=M-1
D=M
A=A-1
M=M+D
@112
D=A
@SP
AM=M+1
A=A-1
M=D
@SP
AM=M-1
D=M
A=A-1
M=M-D
@SP
A=M-1
M=-M
@SP
AM=M-1
D=M
A=A-1
M=M&D
@82
D=A
@SP
AM=M+1
A=A-1
M=D
@SP
AM=M-1
D=M
A=A-1
M=M|D
