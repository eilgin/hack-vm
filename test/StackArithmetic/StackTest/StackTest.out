|  RAM[0]  | RAM[256] | RAM[257] | RAM[258] | RAM[259] |
|       3  |      -1  |     892  |     891  |       0  |
